<launch>
	<node pkg="ldtravocean" type="model_node" name="cart2_node"
		args="$(env SIM_MODEL)" if="$(env IS_SIM)">
		<!-- This will also adjust the model sampling time -->
		<param name="Rate" value="10" />
		<param name="ModelWrap" value="10" />
		<remap from="tauIn" to="tauOut" />
		<remap from="meas" to="meas_sim" />
		<param name="maxThrust" value="9.2" />
		<param name="minThrust" value="-9.2" />
		<param name="publish_world" value="false" />
		<param name="use_noise" value="true" unless="$(env NO_NOISE)" />

		<remap from="imu_model" to="imu" />
		<remap from="fix" to="gps" />
		<remap from="tauAch" to="tauAchSim" />
		<param name="tf_prefix" value="$(env TF_PREFIX)" if="$(env USE_TF_PREFIX)" />
	</node>

	<node pkg="cart2" type="cart2_node" name="cart2_node" if="$(env USE_CART)" machine="controlled" >	
		<param name="PortName" value="/dev/ttyS0" />
		<remap from="tauIn" to="tauOut" />
		<param name="UseRPMControl" value="false" />
		<param name="Kp_rpm" value="0.000035" />
		<param name="Ki_rpm" value="0.0001" />
		<param name="MedianSize" value="3" />
		<remap from="tauAch" to="tauAchDriver" />
        </node>
	

	<node pkg="cart2" type="bench_radio" name="cart2_node" if="$(env USE_BENCH_RADIO)" machine="controlled">
	    <param name="PortName" value="/dev/ttyUSB0" />
		<!--  <param name="PortName" value="/dev/pts/6" /> -->
		<param name="BaudRate" value="19200" />
		<param name="IsBench" value="true" />
		<remap from="tauIn" to="tauOut" />
	</node>
	
	<node pkg="cart2" type="vr_node" name="vr_node" if="$(env USE_VR)" machine="controlled">
	    <param name="PortName" value="/dev/ttyUSB1" />
	    <param name="BaudRate" value="9600" />
	    <param name="an" value="0.0003" />
	    <param name="bn" value="0.0001389" />
	    
	    <remap from="vr_imu" to="imu" />
	    <remap from="tauIn" to="tauOut" />	        
	</node>
</launch>
